ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -190,15 +189,10 @@ class SimpleOperationTracker implements OperationTracker {
 
     // Note that we get a snapshot of replicas by state only once in this class, and use the same snapshot everywhere
     // to avoid the case where a replica state might change in between an operation.
-    allDcReplicasByState =
+    replicasByStateSnapshot =
         (Map<ReplicaState, List<ReplicaId>>) partitionId.getReplicaIdsByStates(EnumSet.allOf(ReplicaState.class), null);
     List<ReplicaId> eligibleReplicas;
     List<ReplicaId> offlineReplicas = new ArrayList<>();
-    totalOfflineReplicaCount =
-        getReplicasByState(null, EnumSet.of(ReplicaState.OFFLINE)).getOrDefault(ReplicaState.OFFLINE,
-            Collections.emptyList()).size();
-    allReplicas = partitionId.getReplicaIds().stream().collect(Collectors.toList());
```

Comment:
When `routerConfig.routerGetEligibleReplicasByStateEnabled`, list of eligible replicas for this operation comes from `partitionId.getReplicaIdsByState` which in turn fetches from helix `RoutingTableSnapshot`. On the other hand, `partitionId.getReplicaIds()` fetches from replica information updated by Ambry in helix property store. I think information in both places could be inconsistent. Due to that, we might need to make sure same source of information is used in all places when calculating counts. This is the main change in this PR.